packages/shorebird_cli/lib/src/commands/patch/ios_patcher.dart

```diff
@@ -81,6 +78,65 @@ class IosPatcher extends Patcher {
     }
   }
 
+  @override
+  Future<DiffStatus> assertUnpatchableDiffs({
+    required ReleaseArtifact releaseArtifact,
+    required File releaseArchive,
+    required File patchArchive,
+  }) async {
+    // Check for diffs without warning about native changes, as Xcode builds
+    // can be nondeterministic. So we still have some hope of alerting users of
+    // unpatchable native changes, we compare the Podfile.lock hash between the
+    // patch and the release.
+    final diffStatus =
+        await patchDiffChecker.confirmUnpatchableDiffsIfNecessary(
+      localArchive: patchArchive,
+      releaseArchive: releaseArchive,
+      archiveDiffer: const IosArchiveDiffer(),
+      allowAssetChanges: allowAssetDiffs,
+      allowNativeChanges: allowNativeDiffs,
+      confirmNativeChanges: false,
+    );
+
+    if (!diffStatus.hasNativeChanges) {
+      return diffStatus;
+    }
+
+    final String? podfileLockHash;
+    if (shorebirdEnv.podfileLockFile.existsSync()) {
+      podfileLockHash = sha256
+          .convert(shorebirdEnv.podfileLockFile.readAsBytesSync())
+          .toString();
+    } else {
+      podfileLockHash = null;
```

Comment:
can't we return early here?
```suggestion
      return diffStatus;
```

Reply:
No—if a release has a podfile lock hash but we don't currently have a podfile lock, we need to warn about that.

Reply:
yeah nvm makes sense 👍